File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn transcribe-nucleutide [nucleotide]
  (case nucleotide
      \G \C
      \C \G
      \T \A
      \A \U
      (->> nucleotide
           (str "Invalid nucleutide:")
           (AssertionError.)
           (throw))))
   

(defn to-rna [dna]
  (->> dna
       (map transcribe-nucleutide)
       (apply str)))

File: go/rna-transcription/rna_transcription.go
// Package strand provides RNA transcription tools
package strand

import "strings"

// ToRNA translates a DNA strand given as string dna to RNA
func ToRNA(dna string) string {
	return strings.Map(transcribeBase, dna)
}

func transcribeBase(b rune) rune {
	switch b {
	case 'G':
		return 'C'
	case 'C':
		return 'G'
	case 'T':
		return 'A'
	case 'A':
		return 'U'
	}
	return '0'
}
